llmfoundry/utils/builders.py

```diff
@@ -244,6 +246,10 @@ def _validate_cfg(icl_cfg: DictConfig):
             icl_cfg.max_seq_len = default_max_seq_len
         if 'batch_size' not in icl_cfg:
             icl_cfg.batch_size = default_batch_size
+        if 'pass_at_k' not in icl_cfg:
+            icl_cfg.pass_at_k = 1
+        if 'num_beams' not in icl_cfg:
+            icl_cfg.num_beams = 20
```

Comment:
this came from a combination of claim in the StarCoder paper and backed by our own analysis/simulation of this choice for pass@1.  Would like a comment to that effect added?

Reply:
yeah, a comment would be good. but i was also wondering if this choice is really necessary to have defaulted everywhere because I would expect it to slow things down quite a bit

Reply:
as a meta question, what is our intention for defaults broadly? Coding guidelines for composer only state they should not be a no-op. 

However, setting num_beams too small produces unreliable results. I would not recommend setting default num_beams = 1 as in the original PR because a user shouldn't trust those results. 

It is also the case that if k is larger, num_beams should perhaps grow (this is open to debate amongst us). So even 20 isn't a uniformly good choice for all k.

Perhaps we shouldn't have a default and instead the user needs to make an explicit choice so they need to think about it?

Reply:
Hmm...what do you mean by "it"? leave it as is or remove default?

Reply:
ah sorry, leave it as is.